### PR TITLE
JPMS Strictly Named Modules

### PR DESCRIPTION
**Pull request by GedMarc**
* JAXB Bump to 2.3.0
* Guice Minimum set to 4.2.2 for JPMS
* JSON Setter - couldn't find merge annotation method on branch 2.9
* JPMS Strictly Named Modules with Hard Exports ONLY

JPMS Considerations
===================
For Private/Protected Field Scanning, The required package should open to com.fasterxml.jackson.databind.
This is how it currently works with automatic module naming, but it is now strictly enforced

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -60,6 +56,42 @@
...
 	</configuration>
       </plugin>
+	  
+		 <plugin>
+			<groupId>org.apache.maven.plugins</groupId>
+			<artifactId>maven-compiler-plugin</artifactId>
+			<version>3.8.0</version>
+			<dependencies>
+				<dependency>
+					<groupId>org.ow2.asm</groupId>
```
Comment (cowtowncoder) [inline pom.xml]: Quick question: why is this needed? It is not a direct dependency of `maven-compiler-plugin`, but I assume comes transitively from somewhere? I don't mind it being there, just want to understand.
I think shade plugin at least depends on it?

Comment (GedMarc) [reply pom.xml]: 3.8.0 was released a bit before ASM, ASM 7.0 is the jpms approved one (also guice 4.2.2 etc). Came across this one while doing cglib/asm constructs.

I didn't want to put it into dependency management though, so I kept it local to the compiler plugin.

File: pom.xml [hunk 2]
```diff
@@ -60,6 +56,42 @@
...
+					<artifactId>asm</artifactId>
+					<version>7.0</version>
+				</dependency>
+			</dependencies>
+		</plugin>
+
+        <plugin>
+            <groupId>org.moditect</groupId>
+            <artifactId>moditect-maven-plugin</artifactId>
+            <version>1.0.0.Beta1</version>
```
[the remark this answers is not in the document]
Comment (GedMarc) [reply pom.xml]: I've had to add this to the enforcer as an exception, it isn't allowing the version through

**Comment by GedMarc** [comment]
Closing for the real one
